controllers/calendar/RecurringEventsController.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace humhub\modules\smartVillage\controllers\calendar;
+
+use DateTime;
+use humhub\modules\calendar\interfaces\CalendarService;
+use humhub\modules\calendar\models\CalendarEntry;
+use humhub\modules\calendar\models\fullcalendar\FullCalendar;
+use humhub\modules\content\components\ActiveQueryContent;
+use humhub\modules\content\components\ContentActiveRecord;
+use humhub\modules\content\models\ContentContainer;
+use humhub\modules\rest\components\BaseContentController;
+use humhub\modules\rest\definitions\ContentDefinitions;
+
+class RecurringEventsController extends BaseContentController
```

Comment:
a bearer token is necessary, but like other calendar entries, we need public recurring ones without bearer token.
is a different extended controller needed? and a check for visibility if no token?

Reply:
Hi @donni106 
We checked the result when visibility was the public but same problem we were facing. when we does not pass the bearer token the data will not get but when I pass the bearer token, then it works fine

Reply:
can you solve this in the same way you build the public endpoints without bearer for calendar, posts, spaces?

Reply:
Yes @donni106 , we can try to resolve this issue

Reply:
Hi @donni106 
We checked and noticed that the events are not visible for guest users (without login) both single and recurring. 

For guest users:

![image](https://user-images.githubusercontent.com/103625795/185057738-d98dade7-7595-41fc-abfb-59c3fc33a195.png)


For login users:

![image](https://user-images.githubusercontent.com/103625795/185058186-cee65747-5a11-4f82-bd66-c16a2df4027a.png)

Reply:
what events are not visible for guests? public ones?

Reply:
Hi @donni106 
Yes, In web, when we set a calendar entry as public and when we are trying to see as a guest, it is not visible.

For example Event03

![image](https://user-images.githubusercontent.com/103625795/189049546-f13f7eeb-a500-4339-8404-5dfd866beb2e.png)

We set event03 as public, but when we are seen as a guest user, it is not visible to us (see the attached screenshot below).

![image](https://user-images.githubusercontent.com/103625795/189049800-e26351c1-6ac1-41dd-8ca4-e2eb8648c273.png)

Reply:
Ok thanks for the explanation. Let's see on a later point if we need to come back to this situation somehow.